core/dynamic-tags/manager.php

```diff
@@ -258,26 +277,68 @@ public function get_tags() {
 	/**
 	 * @since 2.0.0
 	 * @access public
+	 * @deprecated 3.5.0 - Use `$this->register()`.
 	 *
 	 * @param string $class
 	 */
 	public function register_tag( $class ) {
-		/** @var Tag $tag */
-		$tag = new $class();
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function(
+			__METHOD__,
+			'3.5.0',
+			'register'
+		);
+
+		/** @var Base_Tag $tag */
+		$instance = new $class();
```

Comment:
@matipojo 

All the other registrars supply the class instance. The finder is the only component that supplies the class name.

To standardize all the registration methods for all the components we have to change the finder registration to supply class instances instead of class names.

If you prefer class name (for performance), we should change all other registrars (widgets, controls, dynamic tags etc.).

Reply:
OK, we will do it in the future.